plugins/kotlin-dataframe/testData/box/joinWithKinds.kt

```diff
@@ -0,0 +1,44 @@
+import org.jetbrains.kotlinx.dataframe.*
+import org.jetbrains.kotlinx.dataframe.annotations.*
+import org.jetbrains.kotlinx.dataframe.api.*
+import org.jetbrains.kotlinx.dataframe.io.*
+
+fun box(): String {
+    val typed = dataFrameOf("name", "age", "city", "weight")(
+        "Alice", 15, "London", 54,
+        "Bob", 45, "Dubai", 87,
+        "Charlie", 20, "Moscow", null,
+        "Charlie", 40, "Milan", null,
+        "Bob", 30, "Tokyo", 68,
+        "Alice", 20, null, 55,
+        "Charlie", 30, "Moscow", 90,
+    )
+
+    val typed2 = dataFrameOf("name", "origin", "grade", "age")(
```

Comment:
Probably we need here one test for nested tables and their joins

Reply:
+